fava/application.py

```diff
@@ -325,6 +326,17 @@ def api_add_document():
             return api_error('{} already exists.'.format(filepath))
 
         file.save(filepath)
+
+        if request.form.get('bfilename', None):
+            # TODO use next_statement_key to calculate the metadata key to
+            # support multiple statement keys
+            relpath = os.path.relpath(filepath, os.path.dirname(request.form['bfilename']))
+            insert_line_in_file(
+                request.form['bfilename'],
+                int(request.form['blineno'])-1,
+                'statement: "{}"'.format(relpath))
+
+        g.api.changed(force=True)
```

Comment:
And there is a general UI bug here as well: When the Journal get's refreshed subsequently, all filters (via GET-params, like "show metadata") are cleared, which is not what the user intended.